libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -70,6 +70,7 @@ object MapboxMetricsReporter : MetricsReporter {
      */
     @JvmStatic
     fun disable() {
+        removeObserver()
```

Comment:
hm, `removeObserver` should be independent on  enable-disable 🤔make sense?

Reply:
Yeah, that's why we have separate method removeObserver()... But sometimes disable() calls in some places where user have no control (service for example)... so this way is more safety. Wdyt?

Reply:
It seems like there's no way to re-enable a metrics session (should there be?), so it makes sense to clean up all the listeners as well.